menas/ui/components/dataset/datasetDetail.controller.js

```diff
@@ -439,6 +440,47 @@ sap.ui.define([
       this._editScheduleDialog.close();
     },
 
+    _setBusy: function(oCtl) {
+      if(oCtl && oCtl.setBusy && oCtl.setBusyIndicatorDelay) {
+        oCtl.setBusyIndicatorDelay(0);
+        oCtl.setBusy(true);
+      }
+    },
+
+    _clearBusy: function(oCtl) {
+      if(oCtl && oCtl.setBusy) {
+        oCtl.setBusy(false);
+      }
+    },
+
+    _clearBusyPromise: function(oCtl, oPromise) {
+      if(oPromise) {
+        oPromise.always(() => {
+          this._clearBusy(oCtl);
+        })
+      } else {
+        this._clearBusy(oCtl);
+      }
+    },
+    
+    scheduleRunNow: function() {
+      this._setBusy(this._scheduleActionMenu);
```

Comment:
This pattern is repeated 3 times. Maybe we should extract a function for it (along with the private `set/clearBusy` funcitons) in some generic location so we can reuse it across controllers.

Reply:
Gonna extract into a function now.

RE generic location, I would like to create a layer or two of generic (UI5) controllers which the entity ones could then extend. But that's for a separate issue.

Reply:
Good idea!